server/src/main/java/com/flipkart/varadhi/verticles/webserver/WebServerVerticle.java

```diff
@@ -289,30 +516,120 @@ private void setupRouteConfigurators() {
         routeBehaviourConfigurators.put(RouteBehaviour.authorized, authzHandler);
     }
 
+    /**
+     * Configures API routes with the appropriate handlers and behaviors.
+     *
+     * @param router    the router to configure
+     * @param apiRoutes the list of route definitions to configure
+     */
     private void configureApiRoutes(Router router, List<RouteDefinition> apiRoutes) {
-        log.info("Configuring API routes.");
+        log.info("Configuring {} API routes", apiRoutes.size());
         FailureHandler defaultFailureHandler = new FailureHandler();
+
         for (RouteDefinition def : apiRoutes) {
             Route route = router.route().method(def.getMethod()).path(def.getPath());
+
+            // Sort behaviors by order and apply them
             RouteBehaviour[] behaviours = def.getBehaviours().toArray(new RouteBehaviour[0]);
             Arrays.sort(behaviours, Comparator.comparingInt(RouteBehaviour::getOrder));
+
             for (RouteBehaviour behaviour : behaviours) {
-                RouteConfigurator routeConfigurator = routeBehaviourConfigurators.getOrDefault(behaviour, null);
-                if (null != routeConfigurator) {
-                    routeConfigurator.configure(route, def);
+                RouteConfigurator configurator = routeBehaviourConfigurators.get(behaviour);
+                if (configurator != null) {
+                    configurator.configure(route, def);
                 } else {
                     String errMsg = String.format("No RouteBehaviourProvider configured for %s.", behaviour);
                     log.error(errMsg);
                     throw new IllegalStateException(errMsg);
                 }
             }
+
+            // Add pre-handlers
             def.getPreHandlers().forEach(route::handler);
+
+            // Add main handler (blocking or non-blocking)
             if (def.isBlockingEndHandler()) {
                 route.handler(wrapBlockingExecution(vertx, def.getEndReqHandler()));
             } else {
                 route.handler(def.getEndReqHandler());
             }
+
+            // Add failure handler
             route.failureHandler(defaultFailureHandler);
         }
     }
+
+    /**
+     * Sets up event handlers for entity events.
+     *
+     * @return a Future that completes when event handlers are set up
+     */
+    @SuppressWarnings ("unchecked")
+    private Future<Void> setupEventHandlers() {
+        Promise<Void> promise = Promise.promise();
```

Comment:
When a app starts:
- connect all the dbs
- init vertx
- vertx cluster (inter node events will start)
- event processor needs to be init
- start the web server verticles.



move this initialization to probably in its own class.